stable/gcloud-endpoints/templates/deployment.yaml

```diff
@@ -0,0 +1,83 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ template "fullname" . }}
+  labels:
+    app: {{ template "fullname" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: {{ template "fullname" . }}
+    spec:
+      containers:
+        - name: endpoints
+          image: "{{ .Values.image }}"
+          imagePullPolicy: {{ default "" .Values.imagePullPolicy | quote }}
+          args: [
+{{- if (.Values.httpPort) }}
```

Comment:
Due to the dual use of the values in both args and ports, I feel like this works better. I'm not certain how a configmap would make this any cleaner.

Also, the way the httpport value defaults if the sslport or http2port are both empty seems tricky to accomplish in any other way.